daffodil-tdml-processor/src/main/scala/org/apache/daffodil/tdml/processor/DaffodilTDMLDFDLProcessor.scala

```diff
@@ -298,6 +298,13 @@ class DaffodilTDMLDFDLProcessor private (private var dp: DataProcessor) extends
 
   def doParseWithBothApis(dpInputStream: java.io.InputStream, saxInputStream: java.io.InputStream,
     lengthLimitInBits: Long): TDMLParseResult = {
+    //
+    // TDML Tests MUST have a length limit. Otherwise they cannot determine if
+    // there is left-over-data or not without doing more reading from the input stream
+    // so as to be sure to hit end-of-data.
+    //
+    Assert.usage(lengthLimitInBits >= 0)
```

Comment:
Can't TDML tests use the isDefinedForLength(1) thing? Then we don't necessarily have to rely on lengthLimit?

Reply:
Yes, I'm convinced the parse result has to expose the ISDIS for this purpose, rather than trying to snapshot it's position and whether it is "at end" at the end of the parse.

Reply:
Done